### PR TITLE
Two pointer Algorithm using java

### LINKED ISSUE #5023: Two pointer Algorithm using java
## 💥 Proposal

It's the use of two different pointers (usually to keep track of array or string indices) to solve a problem involving said indices with the benefit of saving time and space.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by RuchiGitHu**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

The algorithm basically uses the fact that the input array is sorted. We start the sum of extreme values (smallest and largest) and conditionally move both pointers.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixes #5023

### CODE REVIEW

**Comment by RuchiGitHu** [comment]
> ### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
> Yes
> 
> ### Description
> The algorithm basically uses the fact that the input array is sorted. We start the sum of extreme values (smallest and largest) and conditionally move both pointers.
> 
> ### Checklist
> * [x]  I've read the contribution guidelines.
> * [x]  I've checked the issue list before deciding what to submit.
> * [x]  I've edited the `README.md` and link to my code.
> 
> ## Related Issues or Pull Requests
> fixes #5023

@HarshCasper , @ankitaggarwal23 Please review my PR....

**Review by ankitaggarwal23 — COMMENTED** [review]
- Shift the logic to another function and use main for driver code
- Don't use static input use dynamic inputs from user with proper display message.

File: Java/README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 ## Algorithms
 
+- [Two_Pointer_Algorithm](Algorithms/Two_pointer_Algorithm.java)
```
Comment (ankitaggarwal23) [inline Java/README.md]: Rewrite as ` [Two Pointer Algorithm]`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by RuchiGitHu** [comment]
> * Shift the logic to another function and use main for driver code
> * Don't use static input use dynamic inputs from user with proper display message.

Please review again @ankitaggarwal23

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Kindly use proper code format.

**Comment by RuchiGitHu** [comment]
> Kindly use proper code format.

just now intended...

**Comment by RuchiGitHu** [comment]
> > Kindly use proper code format.
> 
> just now intended...

Please check my code @HarshCasper @ankitaggarwal23

File: Java/Algorithms/Two_pointer_Algorithm.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  System.out.println("Please enter length of the array : ");
+  int l = scan.nextInt();
+  System.out.println("Please enter value the sum you want to be : ");
+  int k = scan.nextInt();
+  int array[] = new int[l];
+  for (int n = 0; n < l - 1; n++) {
+    array[n] = scan.nextInt();
+  }
+  two_pointer obj = new two_pointer(array, k);
+}
```
Comment (ankitaggarwal23) [inline Java/Algorithms/Two_pointer_Algorithm.java]: Give one tab space

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Java/Algorithms/Two_pointer_Algorithm.java (hunk 2 above)
Comment (RuchiGitHu) [reply Java/Algorithms/Two_pointer_Algorithm.java]: updated as raquired.. Please review @ankitaggarwal23 .

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**